Build failure with python 3.8

domoticz fails to build with python 3.8 with the following output
```
[100%] Linking CXX executable domoticz
/home/rclinux/rc-buildroot-test/scripts/instance-0/output/host/opt/ext-toolchain/bin/../lib/gcc/arc-buildroot-linux-uclibc/8.2.1/../../../../arc-buildroot-linux-uclibc/bin/ld: CMakeFiles/domoticz.dir/main/EventsPythonModule.cpp.o: in function `_Py_DECREF.isra.12':
EventsPythonModule.cpp:(.text+0x52): undefined reference to `_Py_Dealloc'
/home/rclinux/rc-buildroot-test/scripts/instance-0/output/host/opt/ext-toolchain/bin/../lib/gcc/arc-buildroot-linux-uclibc/8.2.1/../../../../arc-buildroot-linux-uclibc/bin/ld: EventsPythonModule.cpp:(.text+0x52): undefined reference to `_Py_Dealloc'
/home/rclinux/rc-buildroot-test/scripts/instance-0/output/host/opt/ext-toolchain/bin/../lib/gcc/arc-buildroot-linux-uclibc/8.2.1/../../../../arc-buildroot-linux-uclibc/bin/ld: CMakeFiles/domoticz.dir/hardware/plugins/Plugins.cpp.o: in function `_Py_DECREF.isra.21':
Plugins.cpp:(.text+0x96): undefined reference to `_Py_Dealloc'
/home/rclinux/rc-buildroot-test/scripts/instance-0/output/host/opt/ext-toolchain/bin/../lib/gcc/arc-buildroot-linux-uclibc/8.2.1/../../../../arc-buildroot-linux-uclibc/bin/ld: Plugins.cpp:(.text+0x96): undefined reference to `_Py_Dealloc'
/home/rclinux/rc-buildroot-test/scripts/instance-0/output/host/opt/ext-toolchain/bin/../lib/gcc/arc-buildroot-linux-uclibc/8.2.1/../../../../arc-buildroot-linux-uclibc/bin/ld: CMakeFiles/domoticz.dir/hardware/plugins/PluginProtocols.cpp.o: in function `_Py_DECREF.isra.11':
PluginProtocols.cpp:(.text+0x16): undefined reference to `_Py_Dealloc'
/home/rclinux/rc-buildroot-test/scripts/instance-0/output/host/opt/ext-toolchain/bin/../lib/gcc/arc-buildroot-linux-uclibc/8.2.1/../../../../arc-buildroot-linux-uclibc/bin/ld: CMakeFiles/domoticz.dir/hardware/plugins/PluginProtocols.cpp.o:PluginProtocols.cpp:(.text+0x16): more undefined references to `_Py_Dealloc' follow
collect2: error: ld returned 1 exit status
CMakeFiles/domoticz.dir/build.make:3635: recipe for target 'domoticz' failed
make[3]: *** [domoticz] Error 1
```
From my understanding, this is related to the following change in python 3.8 (https://docs.python.org/3/whatsnew/3.8.html#debug-build-uses-the-same-abi-as-release-build):

> On Unix, C extensions are no longer linked to libpython except on Android and Cygwin. It is now possible for a statically linked Python to load a C extension built using a shared library Python. (Contributed by Victor Stinner in bpo-21536.)
> 
> On Unix, when Python is built in debug mode, import now also looks for C extensions compiled in release mode and for C extensions compiled with the stable ABI. (Contributed by Victor Stinner in bpo-36722.)
> 
> To embed Python into an application, a new --embed option must be passed to python3-config --libs --embed to get -lpython3.8 (link the application to libpython). To support both 3.8 and older, try python3-config --libs --embed first and fallback to python3-config --libs (without --embed) if the previous command fails.
> 
> Add a pkg-config python-3.8-embed module to embed Python into an application: pkg-config python-3.8-embed --libs includes -lpython3.8. To support both 3.8 and older, try pkg-config python-X.Y-embed --libs first and fallback to pkg-config python-X.Y --libs (without --embed) if the previous command fails (replace X.Y with the Python version).
> 
> On the other hand, pkg-config python3.8 --libs no longer contains -lpython3.8. C extensions must not be linked to libpython (except on Android and Cygwin, whose cases are handled by the script); this change is backward incompatible on purpose. (Contributed by Victor Stinner in bpo-36721.)

I don't know how to fix it, so I'm seeking your advice.

I see the same:
`
[ 14%] Linking CXX executable domoticz
/mnt/openwrt/OpenwrtATH79/staging_dir/toolchain-mips_24kc_gcc-8.3.0_musl/lib/gcc/mips-openwrt-linux-musl/8.3.0/../../../../mips-openwrt-linux-musl/bin/ld: /mnt/openwrt/OpenwrtATH79/tmp/cc9IMpCM.ltrans6.ltrans.o: in function `_Py_DECREF.isra.14':
<artificial>:(.text+0x1bb8): undefined reference to `_Py_Dealloc'
collect2: error: ld returned 1 exit status
CMakeFiles/domoticz.dir/build.make:3656: recipe for target 'domoticz' failed
make[5]: *** [domoticz] Error 1
make[5]: Leaving directory '/mnt/openwrt/OpenwrtATH79/build_dir/target-mips_24kc_musl/domoticz-757ce059277883acd648774ae23889cbb5f88f6d'
CMakeFiles/Makefile2:86: recipe for target 'CMakeFiles/domoticz.dir/all' failed
make[4]: *** [CMakeFiles/domoticz.dir/all] Error 2
make[4]: Leaving directory '/mnt/openwrt/OpenwrtATH79/build_dir/target-mips_24kc_musl/domoticz-757ce059277883acd648774ae23889cbb5f88f6d'
Makefile:151: recipe for target 'all' failed
make[3]: *** [all] Error 2
make[3]: Leaving directory '/mnt/openwrt/OpenwrtATH79/build_dir/target-mips_24kc_musl/domoticz-757ce059277883acd648774ae23889cbb5f88f6d'
Makefile:139: recipe for target '/mnt/openwrt/OpenwrtATH79/build_dir/target-mips_24kc_musl/domoticz-757ce059277883acd648774ae23889cbb5f88f6d/.built' failed
make[2]: *** [/mnt/openwrt/OpenwrtATH79/build_dir/target-mips_24kc_musl/domoticz-757ce059277883acd648774ae23889cbb5f88f6d/.built] Error 2
make[2]: Leaving directory '/mnt/openwrt/OpenwrtATH79/feeds/packages/utils/domoticz'
time: package/feeds/packages/domoticz/compile#125.60#3.93#127.69
package/Makefile:111: recipe for target 'package/feeds/packages/domoticz/compile' failed
make[1]: *** [package/feeds/packages/domoticz/compile] Error 2
make[1]: Leaving directory '/mnt/openwrt/OpenwrtATH79'
/mnt/openwrt/OpenwrtATH79/include/toplevel.mk:225: recipe for target 'package/domoticz/compile' failed
make: *** [package/domoticz/compile] Error 2`

--- a/hardware/plugins/DelayedLink.h
+++ b/hardware/plugins/DelayedLink.h
@@ -14,6 +14,44 @@
 #include <frameobject.h>
 #include "../../main/Helper.h"
 
+#if PY_VERSION_HEX >= 0x030800f0
+static inline void
+py3__Py_DECREF(const char *filename, int lineno, PyObject *op)
+{
+	(void)filename; /* may be unused, shut up -Wunused-parameter */
+	(void)lineno; /* may be unused, shut up -Wunused-parameter */
+	_Py_DEC_REFTOTAL;
+	if (--op->ob_refcnt != 0)
+	{
+#ifdef Py_REF_DEBUG
+	if (op->ob_refcnt < 0)
+	{
+		_Py_NegativeRefcount(filename, lineno, op);
+	}
+#endif
+	}
+	else
+	{
+		_Py_Dealloc(op);
+	}
+}
+
+#undef Py_DECREF
+#define Py_DECREF(op) py3__Py_DECREF(__FILE__, __LINE__, _PyObject_CAST(op))
+
+static inline void
+py3__Py_XDECREF(PyObject *op)
+{
+	if (op != NULL)
+	{
+		Py_DECREF(op);
+	}
+}
+
+#undef Py_XDECREF
+#define Py_XDECREF(op) py3__Py_XDECREF(_PyObject_CAST(op))
+#endif
+
 namespace Plugins {
 
 #ifdef WIN32

